src/resolve/resolved-dnstls-openssl.c

```diff
@@ -14,6 +14,13 @@
 #include "resolved-dnstls.h"
 #include "resolved-manager.h"
 
+static void dnstls_error_string(int ssl_error, char *buf, size_t count) {
```

Comment:
I think it is better to return `const char*` which is equivalent to `buf`.
Then, you can inline the call:
```c
#define DNSTLS_ERROR_BUFSIZE 256
#define DNSTLS_ERROR_STRING(error) \
        dnstls_error_string(error, (char[DNSTLS_ERROR_BUFSIZE]){}, DNSTLS_ERROR_BUFSIZE))
...
return log_debug_errno(SYNTHETIC_ERRNO(ECONNREFUSED),
                       "Failed to invoke SSL_do_handshake: %s", DNSTLS_ERROR_STRING(error));
```

Reply:
That's a neat trick. Added both suggestions.